### PR TITLE
Feature/daily snapshot

### PR DESCRIPTION
**Pull request by fillup**

### CODE REVIEW

File: lib/reporting/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+				LatencyDataPoints:   0,
+			}
+		}
+
+		if strings.HasPrefix(entry.ID, domain.TaskTypePing) {
+			// Increment counts
+			nodeEntry.LatencyDataPoints++
+
+			// Update update max/min
+			if entry.Latency > nodeEntry.LatencyMax {
```
Comment (Baggerone) [inline lib/reporting/snapshot.go]: For Billy's sake, you could write a function `getLowerFloat` that would change all these `if`'s to single lines.  :-)

File: db/common.go [hunk 2]
```diff
@@ -130,6 +135,135 @@ func scanTable(tableAlias, dataType string) ([]map[string]*dynamodb.AttributeVal
...
+	//		queryCondition = expression.And(timeCondition, conditionOrs[0], conditionOrs[1])
+	//	} else {
+	//		queryCondition.And(timeCondition, expression.Or(conditionOrs[0], conditionOrs[1], conditionOrs...))
+	//	}
+	//} else {
+	//	queryCondition = timeCondition
+	//}
+
+	prefixCount := len(logTypeIDPrefixes)
+	if prefixCount > 0 {
```
[the remark this answers is not in the document]
Comment (fillup) [reply db/common.go]: nope, removed.

File: lib/reporting/snapshot.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+			if entry.Download < nodeEntry.DownloadMin {
+				nodeEntry.DownloadMin = entry.Download
+			}
+
+			// Increment totals
+			nodeEntry.UploadTotal += entry.Upload
+			nodeEntry.DownloadTotal += entry.Download
+
+			// Calculate average
+			if nodeEntry.SpeedTestDataPoints > 0 {
```
[the remark this answers is not in the document]
Comment (fillup) [reply lib/reporting/snapshot.go]: good point, removed

File: lib/reporting/snapshot.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+			}
+			if entry.Latency < nodeEntry.LatencyMin {
+				nodeEntry.LatencyMin = entry.Latency
+			}
+
+			// Increment totals
+			nodeEntry.LatencyTotal += entry.Latency
+
+			// Calculate average
+			if nodeEntry.LatencyAvg > 0 {
```
[the remark this answers is not in the document]
Comment (fillup) [reply lib/reporting/snapshot.go]: good point, removed

File: lib/reporting/snapshot_test.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	startTime, endTime, err := GetStartEndTimestampsForDate(date)
+	results, err := db.GetDailySnapshotsForRange(startTime, endTime, "")
+	if err != nil {
+		t.Error(err)
+		t.Fail()
+	}
+	if len(results) != 2 {
+		t.Error("Not enough results returned, got ", len(results), "expected 2")
+	}
+
```
[the remark this answers is not in the document]
Comment (fillup) [reply lib/reporting/snapshot_test.go]: It returns an array, not a map, so I cant just check for the element by mac addr

[the remark this answers is not in the document]
Comment (Baggerone) [reply lib/reporting/snapshot_test.go]: oops, sorry

File: lib/reporting/snapshot.go (hunk 1 above)
Comment (fillup) [reply lib/reporting/snapshot.go]: yeah good call, refactored that

**Pull request merged**